sympy/functions/elementary/piecewise.py

```diff
@@ -1172,18 +1172,27 @@ def include(c, x, a):
                 cset = iv - covered
                 if not cset:
                     continue
+                try:
+                    a = cset.inf
+                    incl_a = include(c, x, a)
+                except NotImplementedError:
+                    pass # continue with the given `a`
```

Comment:
If this is catching `NotImplementedError` from `cset.inf` then that is the only thing that should be in the try block:
```
                try:
                    a = cset.inf
                except NotImplementedError:
                    pass # continue with the given `a`
                else:
                    incl_a = include(c, x, a)
```

Reply:
Thanks for the reminder of the `else`-after block.